src/pages/home/report/ReportActionsView.js

```diff
@@ -99,6 +101,17 @@ class ReportActionsView extends React.Component {
             this.updateSortedReportActions(nextProps.reportActions);
             return true;
         }
+        if (nextProps.report.unreadActionCount !== this.props.report.unreadActionCount
+            && nextProps.report.lastMessageTimestamp === this.props.report.lastMessageTimestamp
+            && (!lastItem(nextProps.reportActions) || !lastItem(nextProps.reportActions).loading)
+            && nextProps.report.unreadActionCount !== 0) {
+            // We just  want to set the marker if a message has been marked as unread manually. To avoid moving the
+            // marker when new messages come in, we check whether the last action is loading (which would mean the user
+            // just sent a message themself) and if the timestamps match (which would mean they received a message from
+            // other chat participant).
+            this.newMarkerPosition = (nextProps.report.maxSequenceNumber + 1) - nextProps.report.unreadActionCount;
+            return true;
+        }
```

Comment:
@marcaaron 
Aaaaand this is the ugly monster of ugliness. It works. I hate it. Here's the situation:
- We want to change the marker position when someone sets a comments as unread (changes the number of unreadActions) manually, but NOT when new messages come in.
- It is really hard to differentiate those two situations 😆 
- Ideally, if the total number of actions have not changed with this update, but the unreadActionCount has, that would mean the unread was set manually. The reality? Those come **as two separate updates**.

Options I'm considering:
1. This one.
2. Storing a new variable in ONYX. something like `report.markedAsUnreadAction` that is independent from unreadActionCount and is what gets set when the user clicks "Mark as unread". If that is set (or changes), then the marker goes there. Otherwise, it behaves like now.

I'm not a fan of #2 either, since then `unreadActionCount` would be a lie? Maybe?
LMK what you think 😆

Reply:
What makes it hard to differentiate between these two exactly?

- when someone sets a comment as unread manually
- NOT when new messages come in

Tbh, I'm not sure I understand the problem we're trying to solve yet.

If something is "marked as unread" are we doing this at the server level and updating the `lastRead_*` rNVP associated with that report?

If so, I'd imagine that it'd work something like:

1. User updates the report rNVP in the server and optimistically updates the last read in the front end
2. Let the active report know that it should now update its `initialNewMarkerPosition`

If we were still using something like `PubSub` I'd say to just subscribe to an event in the report view that let's it know it should update. Since we don't have that option we could move the `initialNewMarkerPosition` to be a field on the report object and then just update that instead of creating it once in the constructor.

Reply:
Gotcha! Basically similar to the option 2 I proposed, but storing the whole thing in there. Sounds good! Updated!

Reply:
Please add code comments to explain the context around this logic. The discussion here is helpful, but once the PR is merged, it will be difficult to find.